openwisp_radius/management/commands/base/delete_unverified_users.py

```diff
@@ -27,11 +30,14 @@ def add_arguments(self, parser):
     def handle(self, *args, **options):
         days = now() - timedelta(days=int(options['older_than_days']))
         exclude_methods = str(options['exclude_methods']).split(',')
-        User.objects.filter(
+        for user in User.objects.filter(
             date_joined__lt=days,
             registered_user__isnull=False,
             registered_user__is_verified=False,
-        ).exclude(registered_user__method__in=exclude_methods).delete()
+        ).exclude(registered_user__method__in=exclude_methods):
+            if not RadiusAccounting.objects.filter(username=user.username).exists():
+                user.delete()
+
```

Comment:
I followed the issue description, but can we do this?
```suggestion
User.objects.filter(
            date_joined__lt=days,
            registered_user__isnull=False,
            registered_user__is_verified=False,
        ).exclude(registered_user__method__in=exclude_methods, username__in=RadiusAccounting.objects.all().values_list('username')).delete()

```

Reply:
`RadiusAccounting.objects.all().values_list('username')` this is a very costly database query. 
If I am not wrong, It will fetch username of all RadiusAccounting objects in the database first and then execute the other query. You can check the raw SQL queries to verify this.